core/src/main/java/org/kiwix/kiwixmobile/core/search/adapter/SearchViewHolder.kt

```diff
@@ -30,25 +35,30 @@ sealed class SearchViewHolder<in T : SearchListItem>(containerView: View) :
   class RecentSearchViewHolder(
     override val containerView: View,
     private val onClickListener: (SearchListItem) -> Unit,
+    private val onClickListenerNewTab: (SearchListItem) -> Unit,
     private val onLongClickListener: (SearchListItem) -> Unit
   ) : SearchViewHolder<RecentSearchListItem>(containerView) {
     override fun bind(item: RecentSearchListItem) {
       containerView.setOnClickListener { onClickListener(item) }
+      containerView.list_item_search_new_tab_button
+        .setOnClickListener { onClickListenerNewTab(item) }
       containerView.setOnLongClickListener {
         onLongClickListener(item)
         true
       }
-      (containerView as TextView).text = item.value
+      (containerView.list_item_search_text as TextView).text = item.value
     }
   }
 
   class ZimSearchResultViewHolder(
     override val containerView: View,
-    private val onClickListener: (SearchListItem) -> Unit
+    private val onClickListener: (SearchListItem) -> Unit,
+    private val onClickListenerNewTab: (SearchListItem) -> Unit
   ) : SearchViewHolder<ZimSearchResultListItem>(containerView) {
     override fun bind(item: ZimSearchResultListItem) {
       containerView.setOnClickListener { onClickListener(item) }
-      (containerView as TextView).text = item.value
+      list_item_search_new_tab_button.setOnClickListener { onClickListenerNewTab(item) }
+      (list_item_search_text as TextView).text = item.value
```

Comment:
the cast is unecessary, a relic of the old code that was using findViewById

Reply:
When I remove the cast, it gives the error "Unresolved reference: text".

Reply:
I'll hop on your branch and see whats happening

Reply:
Ah, I didn't pay enough attention to your layout file, my apologies, I only have so much time a day on this project.

Instead of using an include we can just use a textview

Reply:
Do you mean I should use the cast? Or do something else? Sorry, Kotlin's pretty new to me.

Reply:
No worries, I'll comment on the actually appropriate file